file_ops.py

```diff
@@ -32,9 +39,20 @@ def download_file(url, file_name):
     """
     # TODO try catch and log
     # TODO handle for rate limiting
+
     save_to_base = tempfile.mkdtemp()
     save_to = os.path.join(save_to_base, file_name)
-    response = requests.get(url, stream=True)
+
+    if USE_CACHED_FILES:
+        f_path, _ = _get_cached_file(url)
+        if f_path is not None:
+            f_path = os.path.abspath(f_path)
+            os.symlink(f_path, save_to)  # target must be a absolute path
+            logging.info("Using local cache {} --> {}".format(save_to, f_path))
+            return save_to
```

Comment:
again... with the logging ... we need a more concise strategy around try/catch and logging errors/warnings, but assuming success

Reply:
debug?